### PR TITLE
Update mode_auto.cpp

### LINKED ISSUE #20142: Copter: Sudden jump in desired altitude when switching to auto
## Bug report

**Issue details**

Create a mission starting with a takeoff command using TERRAIN altitude frame. Takeoff manually and fly to a location (using guided, loiter, etc.), where the terrain is higher or lower than at takeoff. Now switch to AUTO and observe a sudden jump in desired altitude and climb rate with an agressive reaction from the copter. If the terrain is lower than at takeoff the initial jump is upwards and if the terrain is higher (so when you fly up the mountain), the initial jump goes down which can cause a crash. First encountered this problem during an actual flight, then also reproduced it in SITL.

**Version**
4.1.3, 4.1.5

**Platform**
[  ] All
[  ] AntennaTracker
[ x ] Copter
[  ] Plane
[  ] Rover
[  ] Submarine

**Airframe type**
any copter

**Hardware type**
CubeBlack, SITL

**Logs**
[logs.zip](https://github.com/ArduPilot/ardupilot/files/8117159/logs.zip)

### PR DESCRIPTION
**Pull request by Hasi123**
Change to relative frame to avoid terrain bug with takeoff. Fixes #20142

### CODE REVIEW

**Comment by Hasi123** [comment]
I have tested it yesterday in the simulator and today on my own copter. Works just as expected. Here are some logs (before/after behaviour):
[logs_before_after_fix.zip](https://github.com/ArduPilot/ardupilot/files/8132946/logs_before_after_fix.zip)

File: ArduCopter/mode_auto.cpp [hunk 1]
```diff
@@ -282,6 +282,7 @@ void ModeAuto::takeoff_start(const Location& dest_loc)
         // fall back to altitude above current altitude
         alt_target = copter.current_loc.alt + dest.alt;
     }
+    dest.set_alt_cm(alt_target , Location::AltFrame::ABOVE_HOME);
```
Comment (IamPete1) [inline ArduCopter/mode_auto.cpp:285]:
```suggestion
```
I think this makes it slightly clearer. It changes to always set after the sanity check.

```
    // sanity check
 target
    int32_t alt_target_min_cm = copter.current_loc.alt + (copter.ap.land_complete ? 100 : 0);
    alt_target = MAX(alt_target, alt_target_min_cm);

    // set updated altitude in location structure
    dest.set_alt_cm(alt_target, Location::AltFrame::ABOVE_HOME);
```

**Review by IamPete1 — CHANGES_REQUESTED** [review]

**Comment by tridge** [comment]
assigned to @rmackay9

**Comment by rmackay9** [comment]
I've got an updated fix here https://github.com/ArduPilot/ardupilot/pull/20285.  @Hasi123 would you be able to confirm this resolves the issue you've seen?  Thanks again for finding this important bug!

**Pull request closed without merging**